frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,317 @@
+      requestSizeValue: '1',
+      requestSizeUnit: 'Gi',
+      disableForm: false,
+      showKeyValue: false,
+      nameValuePairs: [['', '']],
+      radios: [
+        {
+          value: 'ReadWriteOnce',
+          title: <span className="form-right-padding">Single User (RWO)</span>,
+        },
+        {
+          value: 'ReadWriteMany',
+          title: <span className="form-right-padding">Shared Access (RWX)</span>,
+        },
+        {
+          value: 'ReadOnlyMany',
+          title: <span className="form-right-padding">Read Only (ROX)</span>,
+        },
+      ],
+      dropdownUnits: {
+        Mi: 'MiB',
+        Gi: 'GiB',
+        Ti: 'TiB',
+      },
+    };
+  }
+
+  handleChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    // this handles storageClass, pvcName, accessMode, size
+    const { name, value } = event.currentTarget;
+    this.setState({ [name]: value } as any);
+  };
+
+  handleNameValuePairs = ({ nameValuePairs }) => {
+    this.setState({ nameValuePairs });
+  };
+
+  handleStorageClass = storageClass => {
+    this.setState({ storageClass });
+  };
+
+  onShowLabelSelectors = () => {
+    this.setState({ showKeyValue: !this.state.showKeyValue });
+  };
+
+  onFormDisable = disableForm => {
+    this.setState({ disableForm });
+  };
+
+  validateForm = () => {
+    const { pvcName } = this.state;
+    // Check that a valid PVC has been choosen
+    if (!pvcName) {
+      this.setState({ error: 'Persistent volume claim name must be set.' });
+      return false;
+    }
+    return true;
+  };
+
+  nameValueArrayToObject = nvpArray => {
+    return _.reduce(nvpArray, (acc, [key, value]) => {
+      return (key !== '' && value !== '') ? { ...acc, [key]: value } : { ...acc };
+    }, {});
+  };
+
+  save = e => {
+    e.preventDefault();
+    if (!this.validateForm()) {
+      return;
+    }
+    const obj = {
+      apiVersion: 'v1',
+      kind: 'PersistentVolumeClaim',
+      metadata: {
+        name: '',
+        namespace: '',
+        annotations: {},
+      },
+      spec: {
+        accessModes: {},
+        resources: {
+          requests: {
+            storage: '',
+          },
+        },
+        selector: {
+          matchLabels: {},
+        },
+      },
+    };
+
+    const { kindObj, namespace } = this.props;
+    let { pvcName, accessMode, requestSizeValue, requestSizeUnit, storageClass, nameValuePairs } = this.state;
+
+    obj.metadata.name = pvcName;
+    obj.metadata.namespace = namespace;
+    obj.spec.accessModes = [accessMode];
+    obj.spec.resources.requests.storage = `${requestSizeValue}${requestSizeUnit}`;
+    //Code for label selectors
+    const valuePairs = this.nameValueArrayToObject(nameValuePairs);
+    if (!_.isEmpty(valuePairs)) {
+      _.set(obj, 'spec.selector.matchLabels', valuePairs);
+    }
+    if (storageClass && storageClass !== 'No Storage Class') {
+      obj.metadata.annotations['volume.beta.kubernetes.io/storage-class'] = storageClass;
```

Comment:
Need to add an option of 'No Storage Class' so user can unselect storage class if they do not want to associate to pvc.